openquakeplatform/openquakeplatform/static/js/risk_input_model_exposure.js

```diff
@@ -0,0 +1,437 @@
+      License, or (at your option) any later version.
+
+      This program is distributed in the hope that it will be useful,
+      but WITHOUT ANY WARRANTY; without even the implied warranty of
+      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+      GNU Affero General Public License for more details.
+
+      You should have received a copy of the GNU Affero General Public License
+      along with this program.  If not, see <https://www.gnu.org/licenses/agpl.html>.
+*/
+
+var table;
+var header;
+var showArea = false;
+
+$( document ).ready(function() {
+    updateTable();
+    $('#outputDiv').css('display', 'none');
+});
+
+$('#retrofittingSelect').hide();
+
+/////////////////////////////////////////////////////////
+// Manage the visibility of the perArea selection menu //
+/////////////////////////////////////////////////////////
+$('#perArea').hide();
+
+$('#defineCostStruc').change(function() {
+    var defineCost = $('#defineCostStruc').val();
+    if (defineCost == 'per_area') {
+        $('#perArea').show();
+        showArea = true;
+    }
+    if (defineCost != 'none') {
+        $('#retrofittingSelect').show();
+    } else {
+        $('#retrofittingSelect').hide();
+        // uncheck retrofitting
+        $('#retroChbx').attr('checked', false);
+    }
+});
+
+$('#defineCostNonStruc').change(function() {
+    var defineCost = $('#defineCostNonStruc').val();
+    if (defineCost == 'per_area') {
+        $('#perArea').show();
+        showArea = true;
+    }
+});
+
+$('#defineCostContent').change(function() {
+    var defineCost = $('#defineCostContent').val();
+    if (defineCost == 'per_area') {
+        $('#perArea').show();
+        showArea = true;
+    }
+});
+
+$('#defineCostBusiness').change(function() {
+    var defineCost = $('#defineCostBusiness').val();
+    if (defineCost == 'per_area') {
+        $('#perArea').show();
+        showArea = true;
+    }
+});
+
+// End
+
+// TODO remove this
+$('#structuralChbx').change(function() {
+    if(this.checked) {
+        $('#retrofittingSelect').show();
+    } else {
+        $('#retrofittingSelect').hide();
+        // uncheck retrofitting
+        $('#retroChbx').attr('checked', false);
+    }
+});
+
+$('#exposureForm').change(function() {
+    updateTable();
+    $('#outputDiv').css('display', 'none');
+});
+
+function checkForValueInHeader(header, argument) {
+    var inx = header.indexOf(argument);
+    return inx;
+}
+
+function updateTable() {
+    // Remove any existing table
+    try {
+        table.destroy();
+    } catch (e) {
+        // continue
+    }
+
+    // Default columns
+    header = ['id', 'longitude', 'latitude', 'taxonomy', 'number'];
+
+    function checkForValue (argument, valueArg) {
+        // Modify the table header only when the menu is altered
+        // This constraint will allow Limit, Deductible and Occupant elements to be
+        // added to the header
+        if (argument != 'none' && valueArg === undefined) {
```

Comment:
Please be sure this check works (see http://stackoverflow.com/questions/28583578/best-practice-of-checking-if-variable-is-undefined). For the reason explained in the top rated answer, I prefer using `typeof valueArg == 'undefined'`, which is safer in general because it avoids the possible reference error when `valueArg` was not declared with `var`. Please check also all the other occurrences of this.